okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java

```diff
@@ -925,96 +925,98 @@ private ModuleDescriptor findSystemInterface(Tenant tenant, String interfaceName
       Map<String, ModuleDescriptor> modsEnabled, InstallJob job) {
 
     List<TenantModuleDescriptor> tml = job.getModules();
-    return DepResolution.installSimulate(modsAvailable, modsEnabled, tml).compose(res -> {
-      if (options.getSimulate()) {
-        return Future.succeededFuture(tml);
-      }
-      return jobs.add(t.getId(), job.getId(), job).compose(res2 -> {
-        Promise<List<TenantModuleDescriptor>> promise = Promise.promise();
-        Future<Void> future = Future.succeededFuture();
-        if (options.getAsync()) {
-          List<TenantModuleDescriptor> tml2 = new LinkedList<>();
-          for (TenantModuleDescriptor tm : tml) {
-            tml2.add(tm.cloneWithoutStage());
-          }
-          promise.complete(tml2);
-        }
-        future = future.compose(x -> {
-          for (TenantModuleDescriptor tm : tml) {
-            tm.setStage(TenantModuleDescriptor.Stage.pending);
+    return DepResolution.installSimulate(modsAvailable, modsEnabled, tml, options.getReinstall())
+        .compose(res -> {
+          if (options.getSimulate()) {
+            return Future.succeededFuture(tml);
           }
-          return jobs.put(t.getId(), job.getId(), job);
-        });
-        if (options.getDeploy()) {
-          future = future.compose(x -> autoDeploy(t, job, modsAvailable, tml));
-        }
-        for (TenantModuleDescriptor tm : tml) {
-          future = future.compose(x -> {
-            tm.setStage(TenantModuleDescriptor.Stage.invoke);
-            return jobs.put(t.getId(), job.getId(), job);
-          });
-          if (options.getIgnoreErrors()) {
-            Promise<Void> promise1 = Promise.promise();
-            installTenantModule(t, pc, options, modsAvailable, tm).onComplete(x -> {
-              if (x.failed()) {
-                logger.warn("Ignoring error for tenant {} module {}",
-                    t.getId(), tm.getId(), x.cause());
+          return jobs.add(t.getId(), job.getId(), job).compose(res2 -> {
```

Comment:
```suggestion
          return jobs.add(t.getId(), job.getId(), job);
        }).compose(res -> {
```
Avoids nesting and reduced indentation.

Reply:
I don't see how this can work here.

Reply:
I've split it into two methods instead.